src/protocols/BaseProtocolAdapter.ts

```diff
@@ -102,32 +102,30 @@ export abstract class BaseProtocolAdapter {
    */
   abstract discoverLastUsedAccountIndex(seed: Uint8Array): Promise<number>;
 
-  abstract constructAndSignTx(
+  constructAndSignTx?(
```

Comment:
I think this should be required for the adapters and we should align the AE adapter to use this method. What do you think @martinkaintas @CedrikNikita ?

Reply:
This is only used by the Bitcoin adapter

Reply:
Since we had a Dogecoin integration planned in future, that can be required (Dogecoin is a fork of Bitcoin).

Reply:
It is only being used by the BTC adapter to calculate byte size of the transaction. I don't think we need this to be required

Reply:
> This is only used by the Bitcoin adapter

Yes, but maybe we should have the signing done on the adapter level? At least as a goal forced by this file.

Reply:
I don't see a clear refactoring path for this method in the `AeternityAdapter` 
I suggest that if we really want this it should be done in a separate PR

Reply:
new task: #2773